client/src/Features/MicrosoftLearn/MicrosoftLearnFilterComponent.tsx

```diff
@@ -0,0 +1,85 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License.
+ *--------------------------------------------------------------------------------------------*/
+
+import React, { useState, useEffect } from 'react';
+import { SimpleComponentStyles } from '../../Core/Utils/FluentUI/typings.fluent-ui';
+import { Text } from '@fluentui/react';
+import { useObserver } from 'mobx-react-lite';
+import { SearchBox } from '@fluentui/react';
+import { FilterComponentProps } from './MicrosoftLearnFilterComponentProps';
+import { getDisplayFromSearch } from './MicrosoftLearnFilterUtils';
+import { getRegexs } from './MicrosoftLearnFilterCore';
+import { MicrosoftLearnFilterItem } from './MicrosoftLearnFilterItem';
+import { useStore } from '../../Stores/Core';
+
+export type MicrosoftLearnFilterStyles = SimpleComponentStyles<'root'>;
+
+const FilterComponentInner = (props: FilterComponentProps): JSX.Element => {
+  const learnStore = useStore('microsoftLearnStore');
+  const [filterSearchTerm, setFilterSearchTerm] = useState('');
+  const [displayOptions, setDisplayOptions] = useState(props.filterOption);
+
+  useEffect(() => {
+    const filterBySearchTerm = (stringExp: string) => {
+      if (stringExp && stringExp?.trim() !== '') {
+        let regexs: RegExp[] = getRegexs(stringExp);
+        if (props.filterOption) {
```

Comment:
I only see an if block inside the if block. Won't clubbing both the if blocks together give the same result?

Reply:
Agreed.